agent/agent.go

```diff
@@ -94,6 +100,86 @@ type outputUnit struct {
 	outputs []*models.RunningOutput
 }
 
+func (a *Agent) runSingleInput(input_config *models.InputConfig, ctx context.Context) {
+	startTime := time.Now()
+
+	plugin := inputs.Inputs[input_config.Name]
+	input := models.NewRunningInput(plugin(), input_config)
+
+	// Overwrite agent interval if this plugin has its own.
+	interval := a.Config.Agent.Interval.Duration
+	if input.Config.Interval != 0 {
+		interval = input.Config.Interval
+	}
+
+	// Overwrite agent precision if this plugin has its own.
+	precision := a.Config.Agent.Precision.Duration
+	if input.Config.Precision != 0 {
+		precision = input.Config.Precision
+	}
+
+	// Overwrite agent collection_jitter if this plugin has its own.
+	jitter := a.Config.Agent.CollectionJitter.Duration
+	if input.Config.CollectionJitter != 0 {
+		jitter = input.Config.CollectionJitter
+	}
+
+	var ticker Ticker
+	if a.Config.Agent.RoundInterval {
+		ticker = NewAlignedTicker(startTime, interval, jitter)
+	} else {
+		ticker = NewUnalignedTicker(interval, jitter)
+	}
+
+	acc := NewAccumulator(input, a.iu.dst)
+	acc.SetPrecision(getPrecision(precision, interval))
+
+	a.wg.Add(1)
+	go func(input *models.RunningInput) {
+		defer ticker.Stop()
+		defer a.wg.Done()
+		a.gatherLoop(ctx, acc, input, ticker, interval)
+	}(input)
+
+	a.Config.Inputs = append(a.Config.Inputs, input)
+}
+
+func (a *Agent) AddInputPluginToAgent(pluginName string) {
```

Comment:
Are any of these functions even necessary? ```AddInputPluginToAgent```, ```AddOutputPluginToAgent```, ```GetAgentInputPlugins``` and ```GetAgentOutputPlugins```?

Reply:
Wait what happened to the function that was printing all of the inputs? That was the function that we wanted. I don't think any of these 4 are relevant to the project